cmd/integration_test.go

```diff
@@ -438,48 +430,77 @@ func TestSmokescreenIntegration(t *testing.T) {
 		t.Run("", func(t *testing.T) {
 			testCase.RandomTrace = rand.Int()
 			resp, err := executeRequestForTest(t, testCase, &logHook)
-			conformResult(t, testCase, resp, err, logHook.AllEntries())
+			isValidProxyResponse(t, testCase, resp, err, logHook.AllEntries())
 		})
 	}
 }
 
+// isValidProxyResponseWithUpstream validates tests cases and expected responses
+// from TestUpstreamProxySmokescreenIntegration. This validates that requests
+// sent to a smokescreen instance with an additional upstream proxy set
+// (proxy chaining) forwards the request to the next proxy hop instead of directly
+// to the proxy target.
+func isValidProxyResponseWithUpstream(t *testing.T, test *TestCase, resp *http.Response, err error, logs []*logrus.Entry) {
```

Comment:
Ditto my earlier suggestion — drop the `is*` prefix if you're not returning a boolean.

Reply:
Changed to `validateProxyResponseWithUpstream`